### PR TITLE
feat(validator): added validator for post /movie endpoint

### PR DESCRIPTION
**Pull request by alikslob**
Why: Our API needs to establish what can be added to our database and what cannot.
What: added validators to our /post route 
details: 
the validators:
-require a title
-limit title length 255 characters
-enforce the year to be greater than 1878 (if included)
-enforce the year to be 4 digits
notes:
tests are included for all 4 requirements

### CODE REVIEW

File: test/validators/movie.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	    expect(result.error.details[0].type).to.eql('string.max');
+    });
+
+  });
+
+  describe('release_year', () => {
+
+    it('is after 1878', () => {
+      const payload = {
+        title: 'Lob Club',
```
Comment (tzaitsev) [inline test/validators/movie.test.js:36]: rofl

File: test/validators/movie.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      };
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path[0]).to.eql('release_year');
+	    expect(result.error.details[0].type).to.eql('number.min');
+    });
+
+    it('is limited to 4 digits', () => {
+      const payload = {
+        title: 'Lobian Rhapsody',
```
Comment (tzaitsev) [inline test/validators/movie.test.js:47]: just a 'heh' on this one.

File: test/validators/movie.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      const payload = {};
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path[0]).to.eql('title');
+	    expect(result.error.details[0].type).to.eql('any.required');
+    });
+
+    it('is less than 255 characters', () => {
+      let lob = 'lob';
+      while (lob.length < 255) {
```
Comment (tzaitsev) [inline test/validators/movie.test.js]: https://www.w3schools.com/jsref/jsref_repeat.asp

**Review by tzaitsev — APPROVED** [review]
Minor nit with the possible use of  https://www.w3schools.com/jsref/jsref_repeat.asp

File: lib/plugins/features/movies/index.js [hunk 4]
```diff
@@ -1,12 +1,15 @@
 'use strict';
-
+const MovieValidator = require('../../../validators/movie');
```
Comment (kylekwong) [inline lib/plugins/features/movies/index.js:2]: nit: add newlines before and after this line

File: test/validators/movie.test.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+describe('movie validator', () => {
+
+  describe('title', () => {
+
+    it('is required', () => {
+      const payload = {};
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path[0]).to.eql('title');
+	    expect(result.error.details[0].type).to.eql('any.required');
```
Comment (kylekwong) [inline test/validators/movie.test.js:16]: nit: make the indentation match the line above this

File: test/validators/movie.test.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    it('is less than 255 characters', () => {
+      const lob = 'lob'.repeat(86);
+      const payload = {
+        title: lob
+      };
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path[0]).to.eql('title');
+	    expect(result.error.details[0].type).to.eql('string.max');
```
Comment (kylekwong) [inline test/validators/movie.test.js:27]: nit: make the indentation match the line above this

File: test/validators/movie.test.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    it('is after 1878', () => {
+      const payload = {
+        title: 'Lob Club',
+        release_year: 1800
+      };
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path[0]).to.eql('release_year');
+	    expect(result.error.details[0].type).to.eql('number.min');
```
Comment (kylekwong) [inline test/validators/movie.test.js:42]: nit: make the indentation match the line above this

File: test/validators/movie.test.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    it('is limited to 4 digits', () => {
+      const payload = {
+        title: 'Lobian Rhapsody',
+        release_year: 18000
+      };
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path[0]).to.eql('release_year');
+	    expect(result.error.details[0].type).to.eql('number.max');
```
Comment (kylekwong) [inline test/validators/movie.test.js:53]: nit: make the indentation match the line above this

**Review by kylekwong — APPROVED** [review]

**Pull request merged**